python/cudf_polars/cudf_polars/dsl/expressions/string.py

```diff
@@ -760,12 +762,28 @@ def do_evaluate(
             # TODO: ignores ambiguous
             format, strict, _, _ = self.options
             col = self.children[0].evaluate(df, context=context)
+            plc_col = col.obj
+            if format is None:
+                # Polars begins inference with the first non null value
```

Comment:
Another short circuiting edge case is `if plc_col.size() == plc.col.null_count():` then we know we'll always just raise. I suppose this would also raise in `plc.copying.get_element` trying to get the first element of an empty column but probably the short circuiting allows us to raise a better error message.